### PR TITLE
Fix Promote buttons in Blaze for Simple Classic

### PR DESCRIPTION
**Pull request by candy02058912**
Fixes https://github.com/Automattic/dotcom-forge/issues/6529 and fixes https://github.com/Automattic/dotcom-forge/issues/6539

## Proposed Changes

| Before  | After |
| :-------------: | :-------------: |
| In Simple Classic wp-admin, clicking on Promote doesn't trigger anything  | Opens the promote campaign creation process |

#### Tools > Advertising

ad

#### Tools > Advertising > Campaigns > Details

again

#### Posts & Pages
image
image

### Prerequisite 1: Simple Classic
To enable Simple Classic, create a Simple Site, and in your sandbox wpsh run:
```
update_blog_option( blogID, 'wpcom_admin_interface', 'wp-admin');
update_blog_option( blogID ,'wpcom_classic_early_release', 1 );
```

### Prerequisite 2: Public site

* Run `install-plugin.sh blaze-dashboard fix/blaze-promote` in your sandbox and sandbox widgets.wp.com for testing, alternatively, locally you can checkout to this branch and cd to `apps/blaze-dashboard` and run `yarn dev --sync`
* Go to `/wp-admin/tools.php?page=advertising` and try all entry points to Promote (see screenshots above)

* See the campaign creation page

Other notes: Instructions on how to create a campaign: https://github.com/Automattic/wp-calypso/pull/89484#issuecomment-2051848954

Test on Self-hosted/Atomic/Simple Calypso to make sure we didn't break anything. In Calypso, instead of opening a link, it will open support docs in help center.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

File: client/my-sites/promote-post-i2/hooks/use-open-promote-widget.ts [hunk 1]
```diff
@@ -17,14 +17,19 @@ export interface Props {
 const useOpenPromoteWidget = ( { keyValue, entrypoint, external }: Props ) => {
 	const { openModal } = useRouteModal( 'blazepress-widget', keyValue );
 	const siteId = useSelector( getSelectedSiteId );
+	const isRunningInJetpack = config.isEnabled( 'is_running_in_jetpack_site' );
+	const isRunningInClassicSimple = useSelector( ( state ) =>
+		getSiteOption( state, siteId, 'is_wpcom_simple' )
+	);
+	const isRunningJetpackOrClassicSimple = isRunningInJetpack || isRunningInClassicSimple;
```
Comment (mmtr) [inline client/my-sites/promote-post-i2/hooks/use-open-promote-widget.ts:24]: Nitpick (non-blocking): We have this check now duplicated in a few places. I wonder if we should centralize it in a common function that's reused everywhere so it's easier to maintain future changes to this logic.

**Review by mmtr — APPROVED** [review]
This tests well for me!

Something unrelated to this PR that I noted though is that, if I quit the promote process I'm not redirected back to where I was before:
- Posts > Click "Promote with Blaze" > (Promote process starts) > Click "Back" > Click "Yes, quit"
  - I land in `/wp-admin/tools.php?page=advertising`, but I was expecting to land in Posts
- Pages > Click "Promote with Blaze" > (Promote process starts) > Click "Back" > Click "Yes, quit"
  - I land in `/wp-admin/tools.php?page=advertising`, but I was expecting to land in Pages
- Tools > Advertising > Campaigns > Details > Click "Promote again" > (Promote process starts) > Click "Back" > Click "Yes, quit"
  - I land in `/wp-admin/tools.php?page=advertising`, but I was expecting to land in the campaign I previously selected

File: client/my-sites/promote-post-i2/hooks/use-open-promote-widget.ts (hunk 1 above)
Comment (candy02058912) [reply client/my-sites/promote-post-i2/hooks/use-open-promote-widget.ts:24]: Yup, let's do that in this followup issue: https://github.com/Automattic/dotcom-forge/issues/6569

**Review by sbarbosa — APPROVED** [review]
The change looks good and worked fine 👍

I have one question here: @candy02058912. 
Will we be using the Public API for simple sites? 
Or are we planning to use the site's API, like Atomic?

If you open the Promote modal, you will notice that the hashbang navigation gets broken (because the modal changes it). 
I tested it, and it worked fine. The only thing we missed was the refresh. If a user refreshes the page, the dashboard will load instead of the modal (probably not a blocker). 

We can change the widget's modal behavior to avoid messing up the URL hash, but right now, it only does this for Jetapck, and the widget thinks you are still in Calypso.

Let me know if you already defined what API to use for classic simple. I can add an additional parameter to the DSP Widget to ignore changes in the URL if we are in simple (the same we do for self-hosted jetpack sites).

**Comment by candy02058912** [comment]
> If you open the Promote modal, you will notice that the hashbang navigation gets broken (because the modal changes it).
I tested it, and it worked fine. The only thing we missed was the refresh. If a user refreshes the page, the dashboard will load instead of the modal (probably not a blocker).

@sbarbosa Nice catch there, I wasn't aware of the refresh issue.
I plan to use https://public-api.wordpress.com for minimal changes. We also did that for Stats so the approaches are consistent.

> Let me know if you already defined what API to use for classic simple. I can add an additional parameter to the DSP Widget to ignore changes in the URL if we are in simple (the same we do for self-hosted jetpack sites).

Because we're using https://public-api.wordpress.com, it's supposed to be the same as what Calypso is using.

File: client/my-sites/promote-post-i2/hooks/use-open-promote-widget.ts (hunk 1 above)
Comment (candy02058912) [reply client/my-sites/promote-post-i2/hooks/use-open-promote-widget.ts:24]: > Something unrelated to this PR that I noted though is that, if I quit the promote process I'm not redirected back to where I was before:

I'll confirm to see if this is the same behavior for self-hosted/atomic sites, if it is, I feel like we can work on keeping behavior consistent and revisit this as a whole together.

Comment (candy02058912) [reply client/my-sites/promote-post-i2/hooks/use-open-promote-widget.ts:24]: Seems like the "Promote with Blaze" button doesn't exist for atomic/self-hosted sites in Posts/Pages
image
image

Can you help confirm @sbarbosa? Thanks 🙏

**Comment by candy02058912** [comment]
Merging this for now, but I'll continue to monitor the discussions in this PR.

**Comment by sbarbosa** [comment]
Thanks for the answer @candy02058912!
It's not a problem to use the Public API. The refresh is not a huge problem, and we can easily fix it by letting the DSP widget know and just avoiding altering the hash.

I can help with the changes in the widget site. Let me know when you advance a bit more in the integration, and we can make the change. But it is not a big priority (it's not breaking anything, jet).

**Comment by candy02058912** [comment]
@sbarbosa I think the integration can be considered done after merging https://github.com/Automattic/wp-calypso/pull/89568. We can create a new Blaze from Simple Classic now.

Would be great if you could help with the changes, but like you said, not a big priority as the Blaze flow can be completed.

**Pull request merged**